crates/forge_analyzer/src/definitions.rs

```diff
@@ -226,9 +233,99 @@ pub fn run_resolver(
         module.visit_with(&mut collector);
     }
 
+    // This loop iterates through env's bodies and recreates an updated set of bodies to satisfy SSA form of IR dump.
+    let mut vars_map: HashMap<VarId, (VarKind, Option<VarId>, bool)> = HashMap::new();
+    for (func_id, body) in environment.bodies_mut().enumerate() {
+        // Builds the variables map to track existing all variables, store their VarId, and check if they have been assigned once (by SSA).
+        for var_id in body.vars.keys() {
+            let var_kind = *body.vars.get(var_id).unwrap();
+            vars_map.insert(var_id, (var_kind, None, false));
+        }
+        // Updates the instruction set with creation of new global reference variables if necessary.
+        for (block_id, block) in body.blocks.iter_mut_enumerated() {
+            for inst in block.iter_insts_mut() {
+                match inst {
+                    Inst::Assign(variable, rvalue) => {
+                        update_rvalue(rvalue, &vars_map);
+                        let var_id = variable.as_var_id().unwrap();
```

Comment:
`variable.as_var_id()` is `None` when `variable` is `super` or `this`. The code should skip over the renaming instead of unwrapping the option